src/policies/conditions.ts

```diff
@@ -111,39 +126,83 @@ export class Condition {
   }
 }
 
-class RPCcondition extends Condition {
-  readonly schema = Joi.object({
-    chain: Joi.string().required(),
+// A helper method for making complex Joi types
+const makeGuard = (
+  schema: Joi.StringSchema | Joi.ArraySchema,
+  types: Record<string, string[]>,
+  parent: string
+) => {
+  Object.entries(types).forEach(([key, value]) => {
+    schema = schema.when(parent, {
+      is: key,
+      then: value,
+    });
+  });
+  return schema;
+};
 
-    method: Joi.string().valid('balanceOf', 'eth_getBalance').required(),
+class TimelockCondition extends Condition {
+  readonly schema = Joi.object({
+    returnValueTest: this.makeReturnValueTest(),
+  });
+}
 
-    parameters: Joi.array(),
+class RpcCondition extends Condition {
+  public readonly RPC_METHODS = ['eth_getBalance', 'balanceOf'];
+  public readonly PARAMETERS_PER_METHOD: Record<string, string[]> = {
+    // TODO: Are these supposed to be defined by using context interface?
+    balanceOf: ['address'],
+    eth_getBalance: ['address'],
+  };
 
-    returnValueTest: Joi.object({
-      comparator: Joi.string().valid('==', '>', '<', '<=', '>=').required(),
-      value: Joi.string(),
-    }),
+  readonly schema = Joi.object({
+    chain: Joi.string()
+      .valid(...this.SUPPORTED_CHAINS)
+      .required(),
+    method: Joi.string()
+      .valid(...this.RPC_METHODS)
+      .required(),
+    parameters: Joi.array().required(),
+    returnValueTest: this.makeReturnValueTest(),
   });
 }
 
 class ContractCondition extends Condition {
-  readonly schema = Joi.object({
-    contractAddress: Joi.string().pattern(new RegExp('^0x[a-fA-F0-9]{40}$')),
-
-    chain: Joi.string().required(),
-
-    standardContractType: Joi.string(),
-
-    functionAbi: Joi.string(),
-
-    method: Joi.string(),
-
-    parameters: Joi.array(),
+  public readonly STANDARD_CONTRACT_TYPES = ['erc20', 'erc721', 'erc1155'];
+  public readonly METHODS_PER_CONTRACT_TYPE: Record<string, string[]> = {
+    erc20: ['balanceOf'],
+    erc721: ['balanceOf', 'ownerOf'],
+    erc1155: ['balanceOf'],
+  };
+  public readonly PARAMETERS_PER_METHOD: Record<string, string[]> = {
+    // TODO: Defined using context interface?
+    balanceOf: ['address'],
+    ownerOf: ['address'],
+  };
 
-    returnValueTest: Joi.object({
-      comparator: Joi.string().valid('==', '>', '<', '<=', '>=').required(),
-      value: Joi.string(),
-    }),
+  private makeMethod = () =>
+    makeGuard(
+      Joi.string(),
+      this.METHODS_PER_CONTRACT_TYPE,
+      'standardContractType'
+    );
+  private makeParameters = () =>
+    makeGuard(Joi.array(), this.PARAMETERS_PER_METHOD, 'method');
+
+  public readonly schema = Joi.object({
+    contractAddress: Joi.string()
+      .pattern(new RegExp('^0x[a-fA-F0-9]{40}$'))
+      .required(),
+    chain: Joi.string()
+      .valid(...this.SUPPORTED_CHAINS)
+      .required(),
+    standardContractType: Joi.string()
+      .valid(...this.STANDARD_CONTRACT_TYPES)
```

Comment:
We'll need to source these standard ABIs from somewhere either in the release, or online.  Since this is intended to run in the browser, etherscan might be a continent place to get contract ABIs from given the address and assuming it's been verified.